src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php

```diff
@@ -15,19 +16,33 @@ public function addEmballageFee()
 
         $items = $woocommerce->cart->get_cart();
 
+        $lastEmballageTaxRateId = null;
         $totalEmballagePrice = 0.00;
         foreach ($items as $values) {
             $product = wc_get_product($values['product_id']);
+            $quantity = $values['quantity'];
             if ($product) {
-                $emballagePrice = $product->get_meta(ProductImport::PRODUCT_EMBALLAGE_PRICE_META_KEY);
-                if ($emballagePrice) {
-                    $totalEmballagePrice += (float) $emballagePrice;
+                if ($product->meta_exists(ProductImport::PRODUCT_EMBALLAGE_PRICE_META_KEY)) {
+                    $emballagePrice = $product->get_meta(ProductImport::PRODUCT_EMBALLAGE_PRICE_META_KEY);
+                    $totalEmballagePrice += (float) ($emballagePrice * $quantity);
```

Comment:
```suggestion
                    $totalEmballagePrice +=  round($emballagePrice*100) * $quantity;
```
We could really count in in cents to not have any rounding errors

Reply:
Changes implemented.